libs/traits/src/lib.rs

```diff
@@ -324,6 +324,38 @@ pub trait PoolUpdateGuard {
 	) -> bool;
 }
 
+/// Trait for get feedback before apply certain changes.
+/// It can be used when you need to ask to a third party or external module if applying a change
+/// that has some effect into the system is something healthy.
+pub trait ChangeGuard {
+	/// Associated pool where evaluate the change.
+	type PoolId;
+
+	/// Identification of a change.
+	type ChangeId;
+
+	/// Change information
+	type Change;
+
+	/// Notify a `change` related to a `pool_id`.
+	/// The caller to this method ask for feedback for the implementation of this trait in order
+	/// be able to semantically proceed successful with that change.
+	/// The change intention will be noted by this method and identified by the returned ChangeId.
+	fn note<C: Into<Self::Change>>(
+		pool_id: Self::PoolId,
+		change: C,
+	) -> Result<Self::ChangeId, DispatchError>;
```

Comment:
why generic over `impl`

Reply:
I tried `impl` before, but I had problems mocking these kinds of methods; I have to investigate it more deeply.

Reply:
Then don't bother. Will compiler will defer the generic anyways. Just the signature looks nicer, but who am I telling this, your code looks gorgeous!

Reply:
I totally agree with the `impl`. Thanks for your comment!

Reply:
`impl` is now supported, but we still need the `'static` lifetime, until #1322